Tests fail on Mac

## Bug report
`make test-short` for master fails on mac with the following:

```
go test -short ./plugins/inputs/fail2ban ./plugins/inputs/hddtemp ./plugins/inputs/dmcache ./internal
# github.com/influxdata/telegraf/plugins/inputs/fail2ban
plugins/inputs/fail2ban/fail2ban_test.go:46: undefined: Fail2ban
plugins/inputs/fail2ban/fail2ban_test.go:50: undefined: execCommand
plugins/inputs/fail2ban/fail2ban_test.go:51: undefined: execCommand
FAIL	github.com/influxdata/telegraf/plugins/inputs/fail2ban [build failed]
# github.com/influxdata/telegraf/plugins/inputs/hddtemp
plugins/inputs/hddtemp/hddtemp_test.go:37: undefined: HDDTemp
FAIL	github.com/influxdata/telegraf/plugins/inputs/hddtemp [build failed]
--- FAIL: TestPerDeviceGoodOutput (0.00s)
        Error Trace:    accumulator.go:258
	            	dmcache_test.go:51
	Error:      	unknown measurement dmcache with tags map[device:cs-1]
        Error Trace:    accumulator.go:258
	            	dmcache_test.go:72
	Error:      	unknown measurement dmcache with tags map[device:cs-2]
        Error Trace:    accumulator.go:258
	            	dmcache_test.go:94
	Error:      	unknown measurement dmcache with tags map[device:all]
--- FAIL: TestNotPerDeviceGoodOutput (0.00s)
        Error Trace:    accumulator.go:258
	            	dmcache_test.go:129
	Error:      	unknown measurement dmcache with tags map[device:all]
--- FAIL: TestErrorDuringGettingStatus (0.00s)
        Error Trace:    dmcache_test.go:155
	Error:      	An error is expected but got nil.
--- FAIL: TestBadFormatOfStatus (0.00s)
        Error Trace:    dmcache_test.go:168
	Error:      	An error is expected but got nil.
FAIL
FAIL	github.com/influxdata/telegraf/plugins/inputs/dmcache	0.009s
--- FAIL: TestCombinedOutputError (0.01s)
        Error Trace:    internal_test.go:96
	Error:      	An error is expected but got nil.
--- FAIL: TestRunError (0.00s)
        Error Trace:    internal_test.go:107
	Error:      	An error is expected but got nil.
FAIL
FAIL	github.com/influxdata/telegraf/internal	0.204s
```

--- a/internal/internal_test.go
+++ b/internal/internal_test.go
@@ -40,6 +40,7 @@ func TestSnakeCase(t *testing.T) {
 var (
 	sleepbin, _ = exec.LookPath("sleep")
 	echobin, _  = exec.LookPath("echo")
+	shell, _    = exec.LookPath("sh")
 )
 
 func TestRunTimeout(t *testing.T) {
@@ -84,13 +85,13 @@ func TestCombinedOutput(t *testing.T) {
 // test that CombinedOutputTimeout and exec.Cmd.CombinedOutput return
 // the same output from a failed command.
 func TestCombinedOutputError(t *testing.T) {
-	if sleepbin == "" {
-		t.Skip("'sleep' binary not available on OS, skipping.")
+	if shell == "" {
+		t.Skip("'sh' binary not available on OS, skipping.")
 	}
-	cmd := exec.Command(sleepbin, "foo")
+	cmd := exec.Command(shell, "-c", "false")
 	expected, err := cmd.CombinedOutput()
 
-	cmd2 := exec.Command(sleepbin, "foo")
+	cmd2 := exec.Command(shell, "-c", "false")
 	actual, err := CombinedOutputTimeout(cmd2, time.Second)
 
 	assert.Error(t, err)
@@ -98,10 +99,10 @@ func TestCombinedOutputError(t *testing.T) {
 }
 
 func TestRunError(t *testing.T) {
-	if sleepbin == "" {
+	if shell == "" {
 		t.Skip("'sleep' binary not available on OS, skipping.")
 	}
-	cmd := exec.Command(sleepbin, "foo")
+	cmd := exec.Command(shell, "-c", "false")
 	err := RunTimeout(cmd, time.Second)
 
 	assert.Error(t, err)

--- a/internal/internal_test.go
+++ b/internal/internal_test.go
@@ -100,7 +100,7 @@ func TestCombinedOutputError(t *testing.T) {
 
 func TestRunError(t *testing.T) {
 	if shell == "" {
-		t.Skip("'sleep' binary not available on OS, skipping.")
+		t.Skip("'sh' binary not available on OS, skipping.")
 	}
 	cmd := exec.Command(shell, "-c", "false")
 	err := RunTimeout(cmd, time.Second)

